File: problem-solving/algorithms/implementation/breaking-the-records/breaking-the-records.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the breakingRecords function below.
def breakingRecords(scores, n):
    max=min=scores[0]
    c1=c2=0
    for i in range(1,n):
        if scores[i]>max:
            c1+=1
            max=scores[i]
        if scores[i]<min:
            c2+=1
            min=scores[i]
    return c1,c2

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())
    scores = list(map(int, input().rstrip().split()))

    result = breakingRecords(scores, n)

    fptr.write(' '.join(map(str, result)))
    fptr.write('\n')

    fptr.close()

File: problem-solving/algorithms/implementation/breaking-the-records/breaking-the-records.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {

    // Complete the breakingRecords function below.
    static int[] breakingRecords(int[] scores, int n) {
        var min = scores[0];
        var max = scores[0];
        var c1 = 0;
        var c2 = 0;
        for (int i = 0; i < n; i++)
        {
            if (scores[i] > max) {
                c1+=1;
                max=scores[i];
            }
            if (scores[i] < min) {
                c2+=1;
                min=scores[i];
            }
        }
        return new int[] {c1, c2};
    }

    static void Main(string[] args) {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine());

        int[] scores = Array.ConvertAll(Console.ReadLine().Split(' '), scoresTemp => Convert.ToInt32(scoresTemp))
        ;
        int[] result = breakingRecords(scores, n);

        textWriter.WriteLine(string.Join(" ", result));

        textWriter.Flush();
        textWriter.Close();
    }
}
